libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -127,7 +127,14 @@ public override async Task ContinueConversationAsync(string botAppId, Conversati
         {
             if (string.IsNullOrWhiteSpace(botAppId))
             {
-                throw new ArgumentNullException(nameof(botAppId));
+                var credentials = (SimpleCredentialProvider)this._credentialProvider;
```

Comment:
I don't believe this code can be merged with this as the solution. There's simply no guarantee this cast can be made in the general case.

Reply:
Hi @cleemullins, 

We refactored the code to avoid the cast because, as you mention, it may not always work.
Instead, we added two new properties (getters and setters) in the interface _ICredentialProvider_ to get the **AppId** from there. 
This is not a Breaking change as [_SimpleCredentialProvider_](https://github.com/Microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Connector/Authentication/SimpleCredentialProvider.cs) is the only class which implements this interface. However, is important to notice that every new class which implements _ICredentialProvider_ must have these new properties.

Please, let us know if you have some more feedback.
Thanks!